gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+    private List<User> userList;
+    private UserServiceImpl userService;
+
+    @BeforeAll
+    void beforeAll() {
+        userDao = mock(UserDaoImpl.class);
+        user = new User();
+        user.setId(ID);
+        user.setFirstName(FIST_NAME);
+        user.setLastName(LAST_NAME);
+        user.setLogin(LOGIN);
+        userValidator = mock(UserValidator.class);
+        pagination = new CustomPagination();
+        pagination.setPage(PAGE);
+        pagination.setSize(SIZE);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+        userList = new ArrayList<>();
+        userList.add(user);
+        userService = new UserServiceImpl(userDao, userValidator, paginationValidator);
+    }
+
+    @Test
+    void testCreate() {
+        User expectedResult = user;
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(true);
+        when(userDao.findByLogin(LOGIN)).thenReturn(Optional.empty());
+        when(userDao.create(user)).thenReturn(user);
+        User actualResult = userService.create(user);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testCreateShouldTrowFirstNameFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowLastNameFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowLoginFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowUserAlreadyExistException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(true);
+        when(userDao.findByLogin(LOGIN)).thenReturn(Optional.of(user));
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(UserAlreadyExistException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testFindAll() {
+        List<User> expectedResult = userList;
```

Comment:
I suppose you can directly use userList and do not create an additional variable

Reply:
It's done